coral/src/app/features/topics/details/documentation/components/DocumentationEditView.tsx

```diff
@@ -0,0 +1,126 @@
+import { useState } from "react";
+import {
+  Alert,
+  Box,
+  Button,
+  Icon,
+  SegmentedControl,
+  SegmentedControlGroup,
+  useToast,
+} from "@aivenio/aquarium";
+import { DocumentationView } from "src/app/features/topics/details/documentation/components/DocumentationView";
+import { DocumentationEditor } from "src/app/features/topics/details/documentation/components/DocumentationEditor";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { parseErrorMsg } from "src/services/mutation-utils";
+import { updateTopicDocumentation } from "src/domain/topic/topic-api";
+import loading from "@aivenio/aquarium/icons/loading";
+
+type DocumentationEditViewProps = {
+  documentation?: string;
+  closeEditView: () => void;
+  topicName: string;
+  topicIdForDocumentation: number;
+};
+
+type ViewMode = "edit" | "preview";
+
+function DocumentationEditView({
+  documentation,
+  closeEditView,
+  topicName,
+  topicIdForDocumentation,
+}: DocumentationEditViewProps) {
+  const queryClient = useQueryClient();
+  const [text, setText] = useState(documentation || "");
+  const [viewMode, setViewMode] = useState<ViewMode>("edit");
+
+  // isloading from mutate is false again as soon as mutation was successful,
+  // but we want to make sure we only remove the loading animation when data is refetched
+  // in the background
+  const [saving, setSaving] = useState(false);
+
+  const toast = useToast();
+
+  const { mutate, isError, error } = useMutation(
+    (markdownString: string) => {
+      setSaving(true);
+      return updateTopicDocumentation({
+        topicName,
+        topicIdForDocumentation,
+        markdownString,
+      });
+    },
+    {
+      onSuccess: () => {
+        queryClient.refetchQueries(["topic-overview"]).then(() => {
```

Comment:
I wonder how we could make it so that the refetch would not happen so imperatively.  Maybe if we:
- add a new entry to the `useTopicDetails` context
- which would set a state in `TopicDetails`
- we add this state as a dependency to `getTopicOverview`
- and we dispatch a state change in this `onSuccess`
- the data is refetched

Maybe this could actually be combined with the implementation of the `useDocumentation` hook I talked about in another comment 🤔

Reply:
Hm, I like that we can trigger this refetches in the component without the need to have information travel up or down the component tree. It would help to add constants for our query keys, but all in all I think it's a very useful thing. 

In this case the `saving` is a inbetween state to make sure we're showing the updated topicOverview with updated documentation and don't have a blink in between, but I feel it's fine that it's the responsibility of this component. 

I could def see to extract that out in a custom hook, though, that makes a lot of sense, will split the responsibility better and also help us for connectors :D

Reply:
This was updated - and I've seen that I don't need the `saving` prop at all 👯  the responsibility for showing "hey, topic details are loading" now stays in TopicDetails!